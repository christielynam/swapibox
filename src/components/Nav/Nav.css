.nav-link {
  height: 40px;
  width: 150px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border-radius: 10px;
  font-size: 15px;
  background-color: white;
  color: #081F29;
  text-decoration: none;
}
  
.active {
  background-color: yellow;
}