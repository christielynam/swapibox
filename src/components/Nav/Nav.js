import React from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const NavBar = styled.div`
  display: flex;
  justify-content: space-evenly;
  padding-top: 25px;
  button {
    height: 40px;
    width: 150px;
    border-radius: 10px;
    font-size: 15px;
  };
`

const Nav = () => {
  return (
    <NavBar>
      <NavLink to='/people'><button>PEOPLE</button></NavLink>
      <NavLink to='/planets'><button>PLANETS</button></NavLink>
      <NavLink to='/vehicles'><button>VEHICLES</button></NavLink>
      <NavLink to='/favorites'><button>FAVORITES [num]</button></NavLink>
    </NavBar>
  )
}

export default Nav;
