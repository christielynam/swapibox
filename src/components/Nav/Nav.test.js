import React from 'react';
import Nav from './Nav';
import { shallow } from 'enzyme'

describe('Nav', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Nav />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})