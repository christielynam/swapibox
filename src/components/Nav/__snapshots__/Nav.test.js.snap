// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/people"
        >
          PEOPLE
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/planets"
        >
          PLANETS
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/vehicles"
        >
          VEHICLES
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/favorites"
        >
          FAVORITES [num]
        </NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "PEOPLE",
          "className": "nav-link",
          "exact": true,
          "onClick": undefined,
          "to": "/people",
        },
        "ref": null,
        "rendered": "PEOPLE",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "PLANETS",
          "className": "nav-link",
          "exact": true,
          "onClick": undefined,
          "to": "/planets",
        },
        "ref": null,
        "rendered": "PLANETS",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "VEHICLES",
          "className": "nav-link",
          "exact": true,
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": "VEHICLES",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "FAVORITES [num]",
          "className": "nav-link",
          "exact": true,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": "FAVORITES [num]",
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  justify-content: space-evenly;
  margin-bottom: 25px;
  padding-top: 25px;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link"
            exact={true}
            to="/people"
          >
            PEOPLE
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link"
            exact={true}
            to="/planets"
          >
            PLANETS
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link"
            exact={true}
            to="/vehicles"
          >
            VEHICLES
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link"
            exact={true}
            to="/favorites"
          >
            FAVORITES [num]
          </NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "PEOPLE",
            "className": "nav-link",
            "exact": true,
            "onClick": undefined,
            "to": "/people",
          },
          "ref": null,
          "rendered": "PEOPLE",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "PLANETS",
            "className": "nav-link",
            "exact": true,
            "onClick": undefined,
            "to": "/planets",
          },
          "ref": null,
          "rendered": "PLANETS",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "VEHICLES",
            "className": "nav-link",
            "exact": true,
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "VEHICLES",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "FAVORITES [num]",
            "className": "nav-link",
            "exact": true,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "FAVORITES [num]",
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  justify-content: space-evenly;
  margin-bottom: 25px;
  padding-top: 25px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
